pkg/controller/controller.go

```diff
@@ -149,6 +150,7 @@ func (c *Controller) Run() error {
 				// TODO: set version in spec in apiserver
 				s.Version = defaultVersion
 			}
+			clus.Spec.Version = strings.Trim(clus.Spec.Version, "v")
```

Comment:
> spec should be "readonly"

It depends.
Spec is "readonly" from the perspective of end user. However, admission controller can modify spec.
Having a centralized admission controller style is good for maintaining the assumptions on spec.
Because admission controller isn't extensible yet. We can only do it this way. All spec changes should be processed in centralized way right after passed from watch chan.